profile-service/src/main/java/com/telecominfraproject/wlan/profile/controller/ProfileController.java

```diff
@@ -75,6 +76,14 @@ public Profile create(@RequestBody Profile profile ) {
             LOG.error("Failed to create Profile, request contains unsupported value: {}", profile);
             throw new DsDataValidationException("Profile contains unsupported value");
         }
+        
+        // Check that there are no other profiles of the same type with the same name
+        if (!getForCustomer(profile.getCustomerId(), profile.getProfileType(), profile.getName(), null, null)
```

Comment:
Why at this level? It does not cover the update operation, and it may not work for concurrent requests anyway. 

A better way is to add unique index in the rdbms like this:
create unique index if not exists idx_profile_customerId_profileType_name on profile (customerId, profileType, name);

And for the in-memory datastore we can do the explicit check.